okhttp/src/main/java/com/squareup/okhttp/internal/spdy/SpdyConnection.java

```diff
@@ -791,90 +833,51 @@ public void pushPromise(int streamId, int promisedStreamId, List<Header> request
     }
   }
 
-  /** Even, positive numbered streams are pushed streams in HTTP/2. */
+  /** Even, positive numbered streams are pushed streams in HTTP/2 and SPDY/3. */
   private boolean pushedStream(int streamId) {
-    return protocol == Protocol.HTTP_2 && streamId != 0 && (streamId & 1) == 0;
+    return (protocol == Protocol.HTTP_2 || client) && streamId != 0 && (streamId & 1) == 0;
```

Comment:
Also this change was in the original voxer branch:
see: https://github.com/Voxer/okhttp/commit/683dd50542113500aca4003c750003eff414863b#diff-feaa331049f72c00f6a893aa3e45499dR781.
It does seem to violate the specs though.